### DISCUSSION TITLE
Wazuh Email Alerts

### DISCUSSION THREAD

**Discussion by DC-Smash** [wazuh/wazuh#25379] [General]
Hey everyone.

I was hoping to find some clarity on this topic. I recently set up a postfix relay in order to get specific alerts from Waz. Everything works but the body of the email is a mess. I've read that you can integrate custom formatting scripts to run when a specific alert is triggered but I have tested this and nothing has changed.

Does anyone know how I can format the body of a relayed email from postfix so that it's user friendly? Any info is welcomed!

**Comment by Hasitha9796** [comment]
Hi [DC-Smash](https://github.com/DC-Smash)

You can follow the direct integration method to have Wazuh alerts on email.
https://documentation.wazuh.com/current/user-manual/manager/integration-with-external-apis.html

Here are the steps to achieve.
Wazuh email alerts do not support SMTP servers with authentication, such as Gmail. However, you can send these emails via a server relay like Postfix. Please follow the steps to configure Postfix first.
https://documentation.wazuh.com/current/user-manual/manager/alert-management.html#smtp-server-with-authentication
After following the steps mentioned document, you can follow the next steps.
Create a Python file `custom-email.py` in `/var/ossec/integrations` directory and copy the content from the attached script file.
Set the permission
`chown root:wazuh /var/ossec/integrations/custom-email.py`
`chmod 750 /var/ossec/integrations/custom-email.py`

[custom-email.txt](https://github.com/user-attachments/files/21227493/custom-email.txt)

Add the following block in your manager’s configuration:
```
<integration>
 <name>custom-email.py</name>
 <rule_id>553</rule_id>
 <alert_format>json</alert_format>
 <options>JSON</options>
</integration>
```

Make sure to replace the values in the script. set both as 3 step email while you configure postfix.
```
SENDER_EMAIL = 'username@test.xyz'  #SENDER EMAIL ADDRESS
RECEIVER_EMAIL = 'username1@test.xyz'  #RECEIVER EMAIL ADDRESS
```

Restart the Wazuh manager service:
`systemctl restart wazuh-manager`

Additionally, please check the response in the Discord post on the same topic
https://discord.com/channels/1049711339578331186/1049711340316541004/1392383964546596914

Let me know if you need further assistance on this.
